# HG changeset patch
# User Michael Goffioul <michael.goffioul@gmail.com>
# Date 1377298122 14400
#      Fri Aug 23 18:48:42 2013 -0400
# Node ID c8809de193a3e9a8ea40bff0eb6b224869b989df
# Parent  aa662d2e5829372a047dafe505f74b2872fe70c4
Handle 8-bits characters correctly in freetype renderer.

* libinterp/corefcn/txt-eng-ft.cc
(ft_render::visit(text_element_string)): Cast std::string characters to
"unsigned char" before being implicitly casted to FT_ULong.

diff --git a/libinterp/corefcn/txt-eng-ft.cc b/libinterp/corefcn/txt-eng-ft.cc
--- a/libinterp/corefcn/txt-eng-ft.cc
+++ b/libinterp/corefcn/txt-eng-ft.cc
@@ -633,17 +633,18 @@ ft_render::visit (text_element_string& e
 {
   if (font.is_valid ())
     {
       std::string str = e.string_value ();
       FT_UInt glyph_index, previous = 0;
 
       for (size_t i = 0; i < str.length (); i++)
         {
-          glyph_index = process_character (str[i], previous);
+          glyph_index = process_character (static_cast<unsigned char> (str[i]),
+                                           previous);
 
           if (str[i] == '\n')
             previous = 0;
           else
             previous = glyph_index;
         }
     }
 }
